qiskit/algorithms/linear_solvers/hhl.py

```diff
@@ -307,7 +307,9 @@ def _calculate_observable(
         return result, expectation_results
 
     def construct_circuit(
-        self, matrix: Union[np.ndarray, QuantumCircuit], vector: Union[np.ndarray, QuantumCircuit]
+        self,
+        matrix: Union[list, np.ndarray, QuantumCircuit],
+        vector: Union[list, np.ndarray, QuantumCircuit],
```

Comment:
I think the builtin 'list as a typehint is only for 3.9 and above. Since we still support 3.6+ we use 'List' as imported from typing.